Mathlib/Analysis/Complex/HasPrimitives.lean

```diff
@@ -0,0 +1,435 @@
+/-
+Copyright (c) 2024 Ian Jauslin and Alex Kontorovich. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Ian Jauslin, Alex Kontorovich
+-/
+import Mathlib.Analysis.Complex.CauchyIntegral
+import Mathlib.Analysis.Complex.Convex
+
+/-!
+# Primitives of Holomorphic Functions
+
+In this file, we give conditions under which holomorphic functions have primitives. The main goal
+is to prove that holomorphic functions on simply connected domains have primitives. As a first step,
+we prove that holomorphic functions on discs have primitives. The approach is based on Morera's
+theorem, that a continuous function (on a disc) whose `RectangleIntegral` vanishes on all
+rectangles contained in the disc has a primitive. (Coupled with the fact that holomorphic functions
+satisfy this property.) To prove Morera's theorem, we first define the `WedgeInt`, which is the
+integral of a function over a "wedge" (a horizontal segment followed by a vertical segment in the
+disc), and compute its derivative.
+
+## Main results
+
+* `VanishesOnRectanglesInDisc.diff_of_wedges`: If a function `f` vanishes on all rectangles in a
+  disc with center `c`, then the wedge integral from `c` to `w` minus the wedge integral from
+  `c` to `z` is equal to the wedge integral from `z` to `w`.
+
+* `deriv_of_wedgeInt`: The derivative of the wedge integral is the function being integrated.
+
+* `VanishesOnRectanglesInDisc.hasPrimitive`: **Morera's Theorem**: A function which is
+  continuous on a disc and whose integral on rectangles in the disc vanishes has a primitive
+  on the disc (defined by the wedge integral).
+
+* `hasPrimitives_on_disc`: A holomorphic function on a disc has primitives.
+
+## Tags
+  Holomorphic functions, primitives
+
+TODO: Extend to holomorphic functions on simply connected domains. (In particular, this allows one
+to define the complex logarithm of a nonvanishing function on a simply connected domain.)
+-/
+
+open Complex Topology Set Metric
+
+set_option autoImplicit true
```

Comment:
Could you remove this? Mostly, it confuses people.

Reply:
Hmm why did I think I needed that? Looks like I don't. It's gone. Thanks!